db/seeds.rb

```diff
@@ -6,3 +6,16 @@
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
 
+if ENV['USE_DEMO_DATA'] || Settings.use_demo_data || Rails.env.test?
```

Comment:
I see what you mean.

In order to avoid code duplication with an additional endpoint, as you say, I'd suggest to have a different endpoint in the PlaygroundController that handles the demo data generation, and returns an array of json or xml demo data to the playground page, which will then send it back to the MessagesController one by one as usual.

If the number of tests generated is not too big, it won't be a problem. And on the other hand, if you want to generate a considerable number of tests then you should run the rake task, otherwise you will hang the server and it won't matter the way we implemented the playground generation.